src/screens/Analytics/AnalyticsNew.res

```diff
@@ -0,0 +1,714 @@
+module MetricsState = {
+  open LogicUtils
+  open Promise
+  @react.component
+  let make = (
+    ~singleStatEntity,
+    ~filterKeys,
+    ~startTimeFilterKey,
+    ~endTimeFilterKey,
+    ~moduleName,
+    ~initialFilters,
+    ~options,
+    ~initialFixedFilters,
+    ~tabKeys,
+    ~filterUri,
+    ~heading,
+    ~formaPayload: option<DynamicSingleStat.singleStatBodyEntity => string>=?,
+  ) => {
+    let updateDetails = APIUtils.useUpdateMethod()
+    let defaultFilters = [startTimeFilterKey, endTimeFilterKey]
+    let (filterDataJson, setFilterDataJson) = React.useState(_ => None)
+    let {updateExistingKeys, filterValueJson} = React.useContext(FilterContext.filterContext)
+    let filterData = filterDataJson->Option.getOr(Dict.make()->JSON.Encode.object)
+
+    let setInitialFilters = HSwitchRemoteFilter.useSetInitialFilters(
+      ~updateExistingKeys,
+      ~startTimeFilterKey,
+      ~endTimeFilterKey,
+      ~origin="analytics",
+      (),
+    )
+
+    React.useEffect0(() => {
+      setInitialFilters()
+      None
+    })
+
+    let startTimeVal = filterValueJson->getString("startTime", "")
+    let endTimeVal = filterValueJson->getString("endTime", "")
+
+    let filterBody = React.useMemo3(() => {
+      let filterBodyEntity: AnalyticsUtils.filterBodyEntity = {
+        startTime: startTimeVal,
+        endTime: endTimeVal,
+        groupByNames: tabKeys,
+        source: "BATCH",
+      }
+      AnalyticsUtils.filterBody(filterBodyEntity)
+    }, (startTimeVal, endTimeVal, tabKeys->Array.joinWith(",")))
+
+    React.useEffect3(() => {
+      setFilterDataJson(_ => None)
+      if startTimeVal->LogicUtils.isNonEmptyString && endTimeVal->LogicUtils.isNonEmptyString {
+        try {
+          switch filterUri {
+          | Some(filterUri) =>
+            updateDetails(filterUri, filterBody->JSON.Encode.object, Post, ())
+            ->thenResolve(json => setFilterDataJson(_ => json->Some))
+            ->catch(_ => resolve())
+            ->ignore
+          | None => ()
+          }
+        } catch {
+        | _ => ()
+        }
+      }
+      None
+    }, (startTimeVal, endTimeVal, filterBody->JSON.Encode.object->JSON.stringify))
```

Comment:
api is getting called thrice if we apply some filters

https://github.com/juspay/hyperswitch-control-center/assets/120017870/1ebaab4a-4355-4822-87bd-f81c4b0e2982
try writing separate useEffect for filterbody